core/src/main/java/io/grpc/util/MutableHandlerRegistry.java

```diff
@@ -64,6 +67,12 @@ public boolean removeService(ServerServiceDefinition service) {
     return services.remove(service.getServiceDescriptor().getName(), service);
   }
 
+  @Override
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2222")
+  public List<ServerServiceDefinition> getServices() {
+    return Collections.unmodifiableList(new ArrayList<ServerServiceDefinition>(services.values()));
```

Comment:
Put a comment here that this doesn't necessarily return a consistent view of the map. That's fine because it is "safe" and no ConcurrentModificationException is thrown by ConcurrentHashMap, but it's subtle so it's probably good to call it out in one way or another. If it were a different map implementation, you would probably have needed to use `services.values().toArray()`.